test/indexer/test_indexer.py

```diff
@@ -436,7 +436,7 @@ def test_zero_tallies(self):
         for doc_id in doc_ids:
             message_re = re.compile(fr'INFO:elasticsearch:'
                                     fr'Got 201 response after [^ ]+ from PUT to '
-                                    fr'.*_aggregate/_doc/{doc_id}.*')
+                                    fr'.*_aggregate/_create/{doc_id}.*')
```

Comment:
> I don't understand how the optype ends up as part of the URL, in place of _doc

The request URL is composed within the various methods of the `Elasticsearch` instance. The details of how the URL is composed depend on which method we call.

On `develop`, we use the same method, `Elasticsearch.index`, to perform both `create` and `index` operations, and use the `op_type` query parameter for `create` operations. This method always uses "_doc" as the default document type when none is explicitly provided, regardless of the `op_type` query parameter. [source](https://github.com/elastic/elasticsearch-py/blob/a3b7d2d8ac33245e9e2f4ca2099d959368f1463b/elasticsearch/client/__init__.py#L396).

On this branch, we retain the use `ElasticSearch.index` for `index` operations, but use `Elasticsearch.create` instead for `create` operations instead of specifying an `op_type`. This method always uses "_create" as the default document type when none is explicitly provided, and does not support the `op_type` query parameter. [source](https://github.com/elastic/elasticsearch-py/blob/a3b7d2d8ac33245e9e2f4ca2099d959368f1463b/elasticsearch/client/__init__.py#L332).

Reply:
> Ideally, a refactoring like this shouldn't alter the semantics but this seems like a semantic change.

The `.../_doc/...` and `.../_create/...` endpoints share the same documentation:

https://www.elastic.co/guide/en/elasticsearch/reference/7.17/docs-index_.html#docs-index-api-desc
> You can index a new JSON document with the _doc or _create resource. Using _create guarantees that the document is only indexed if it does not already exist. To update an existing document, you must use the _doc resource.

https://www.elastic.co/guide/en/elasticsearch/reference/7.17/docs-index_.html#operation-type
> You can force a create operation by using the _create resource or setting the op_type parameter to create.

I interpret these excerpts to mean that  `PUT .../create/...` is semantically equivalent to `PUT .../index/...?op_type=create`, so I don't believe that this is a semantic change.

Reply:
I see. That explains the change. When we talk about semantic changes of refactoring it is usually referring to the program being refactored itself. Since Azul and Elasticsearch are different programs, changing one program to hit a different endpoint in another constitutes a semantic change but it is one I am OK with in this case.